File: javascript/difference-of-squares/difference-of-squares.js
  class Squares {
    constructor(num) {
      this.num = num;
    }
     

    
  }

  Squares.prototype.squareOfSum = function() {
    let squareOfSumArray = [];
    for (let i = 1; i <= this.num; i++) {
    squareOfSumArray.push(i);
    }
    return Math.pow(squareOfSumArray.reduce((a, b) => a + b, 0), 2);
  }

  Squares.prototype.sumOfSquares = function() {
    let sumOfSquaresArray = [];
    for (let i = 1; i <= this.num; i++) {
      sumOfSquaresArray.push(i * i);
    }

    return sumOfSquaresArray.reduce((a, b) => a + b, 0);
  }

  Squares.prototype.difference = function() {
     return this.squareOfSum() - this.sumOfSquares();
  }
  


export default Squares;

let squares5 = new Squares(100);

console.log(squares5.sumOfSquares());

console.log(squares5.squareOfSum());

console.log(squares5.difference());

File: ruby/difference-of-squares/difference_of_squares.rb


class Squares
  def initialize user_input
    @user_input = user_input
  end

  def square_of_sum
    result = 0

    for i in 1..@user_input do
      result += i
    end
    result**2
  end

  def sum_of_squares
    result = 0
    for i in 1..@user_input do
      result += i**2
    end
    result
  end

  def difference
    square_of_sum - sum_of_squares
  end
end